src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ActionPlan.kt

```diff
@@ -35,4 +35,9 @@ data class ActionPlan(
   // Required
   @NotNull @OneToOne @Fetch(FetchMode.JOIN) val referral: Referral,
   @Id val id: UUID,
-)
+) {
+  // This is to avoid stack overflow issues with the bi-directional association with referral
+  override fun toString(): String {
+    return "ActionPlan(referralId=${referral.id}, numberOfSessions=$numberOfSessions, activities=$activities, createdBy=$createdBy, createdAt=$createdAt, submittedBy=$submittedBy, submittedAt=$submittedAt, id=$id)"
```

Comment:
i understand the reason this exists, but shouldn't it be JSON?

Reply:
The toString is as per Kotlin which isn't strictly json - e.g. AuthUser(id=referral_repository_test_user_id, authSource=delius, userName=bernard.beaks)